packages/core/content-releases/server/src/services/release.ts

```diff
@@ -230,6 +231,38 @@ const createReleaseService = ({ strapi }: { strapi: LoadedStrapi }) => ({
 
     return contentTypesMeta;
   },
+  async delete(releaseId: DeleteRelease.Request['params']['id']) {
+    const release = (await strapi.entityService.findOne(RELEASE_MODEL_UID, releaseId, {
+      populate: {
+        actions: {
+          fields: ['id'],
+        },
+      },
+    })) as unknown as Release;
+
+    if (!release) {
+      throw new errors.NotFoundError(`No release found for id ${releaseId}`);
+    }
+
+    if (release.releasedAt) {
+      throw new errors.ValidationError('Release already published');
+    }
+
+    // Only delete the release and its actions is you in fact can delete all the actions and the release
+    // Otherwise, if the transaction fails it throws an error
+    await strapi.db.transaction(async () => {
+      await strapi.db.query(RELEASE_ACTION_MODEL_UID).deleteMany({
+        where: {
+          id: {
+            $in: release.actions.map((action) => action.id),
+          },
+        },
+      });
+      await strapi.entityService.delete(RELEASE_MODEL_UID, releaseId);
```

Comment:
I notice here you perform the delete of the release inside the transaction, whereas for update you update the releas outside the transaction. What's the difference?

Reply:
I guess you mean when publishing a release...

Actually it's a good point, in the publishing case makes more sense put that update inside the transaction 🤔, so we make sure we don't publish entries if the update fails (it's a unlikely case but I guess is possible) 

Are you okay if I make that change in this same PR?

Reply:
Hey yeah that's fine by me to make the same update here in this PR if you want otherwise this one looks good and we can merge.